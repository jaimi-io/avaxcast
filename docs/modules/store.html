<!DOCTYPE html>
<html class="default no-js">
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>store | avaxcast-frontend</title>
    <meta name="description" content="Documentation for avaxcast-frontend" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="../assets/css/main.css" />
    <script async src="../assets/js/search.js" id="search-script"></script>
  </head>
  <body>
    <header>
      <div class="tsd-page-toolbar">
        <div class="container">
          <div class="table-wrap">
            <div
              class="table-cell"
              id="tsd-search"
              data-index="../assets/js/search.json"
              data-base=".."
            >
              <div class="field">
                <label
                  for="tsd-search-field"
                  class="tsd-widget search no-caption"
                  >Search</label
                >
                <input id="tsd-search-field" type="text" />
              </div>
              <ul class="results">
                <li class="state loading">Preparing search index...</li>
                <li class="state failure">The search index is not available</li>
              </ul>
              <a href="../index.html" class="title">avaxcast-frontend</a>
            </div>
            <div class="table-cell" id="tsd-widgets">
              <div id="tsd-filter">
                <a
                  href="#"
                  class="tsd-widget options no-caption"
                  data-toggle="options"
                  >Options</a
                >
                <div class="tsd-filter-group">
                  <div class="tsd-select" id="tsd-filter-visibility">
                    <span class="tsd-select-label">All</span>
                    <ul class="tsd-select-list">
                      <li data-value="public">Public</li>
                      <li data-value="protected">Public/Protected</li>
                      <li data-value="private" class="selected">All</li>
                    </ul>
                  </div>
                  <input type="checkbox" id="tsd-filter-inherited" checked />
                  <label class="tsd-widget" for="tsd-filter-inherited"
                    >Inherited</label
                  >
                  <input type="checkbox" id="tsd-filter-externals" checked />
                  <label class="tsd-widget" for="tsd-filter-externals"
                    >Externals</label
                  >
                </div>
              </div>
              <a href="#" class="tsd-widget menu no-caption" data-toggle="menu"
                >Menu</a
              >
            </div>
          </div>
        </div>
      </div>
      <div class="tsd-page-title">
        <div class="container">
          <ul class="tsd-breadcrumb">
            <li>
              <a href="../modules.html">avaxcast-frontend</a>
            </li>
            <li>
              <a href="store.html">store</a>
            </li>
          </ul>
          <h1>Module store</h1>
        </div>
      </div>
    </header>
    <div class="container container-main">
      <div class="row">
        <div class="col-8 col-content">
          <section class="tsd-panel-group tsd-index-group">
            <h2>Index</h2>
            <section class="tsd-panel tsd-index-panel">
              <div class="tsd-index-content">
                <section class="tsd-index-section">
                  <h3>Type aliases</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#AppDispatch" class="tsd-kind-icon"
                        >App<wbr />Dispatch</a
                      >
                    </li>
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#RootState" class="tsd-kind-icon"
                        >Root<wbr />State</a
                      >
                    </li>
                  </ul>
                </section>
                <section class="tsd-index-section">
                  <h3>Variables</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#default" class="tsd-kind-icon"
                        >default</a
                      >
                    </li>
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#persistor" class="tsd-kind-icon"
                        >persistor</a
                      >
                    </li>
                  </ul>
                </section>
              </div>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Type aliases</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="AppDispatch" class="tsd-anchor"></a>
              <h3>App<wbr />Dispatch</h3>
              <div class="tsd-signature tsd-kind-icon">
                App<wbr />Dispatch<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.dispatch</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L10"
                      >src/store.ts:10</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="RootState" class="tsd-anchor"></a>
              <h3>Root<wbr />State</h3>
              <div class="tsd-signature tsd-kind-icon">
                Root<wbr />State<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">ReturnType</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.getState</span
                ><span class="tsd-signature-symbol">&gt;</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L9"
                      >src/store.ts:9</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Variables</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="default" class="tsd-anchor"></a>
              <h3><span class="tsd-flag ts-flagConst">Const</span> default</h3>
              <div class="tsd-signature tsd-kind-icon">
                default<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Store</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-type">EmptyObject</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-symbol">{ </span>isDark<span
                  class="tsd-signature-symbol"
                  >: </span
                ><span class="tsd-signature-type">boolean</span
                ><span class="tsd-signature-symbol">; </span>marketFilter<span
                  class="tsd-signature-symbol"
                  >: </span
                ><a
                  href="../enums/common_enums.Market.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Enumeration"
                  >Market</a
                ><span class="tsd-signature-symbol"> }</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-type">PersistPartial</span
                ><span class="tsd-signature-symbol">, </span
                ><a
                  href="../interfaces/lib_global.Action.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Interface"
                  >Action</a
                ><span class="tsd-signature-symbol">&gt;</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L5"
                      >src/store.ts:5</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="persistor" class="tsd-anchor"></a>
              <h3>
                <span class="tsd-flag ts-flagConst">Const</span> persistor
              </h3>
              <div class="tsd-signature tsd-kind-icon">
                persistor<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Persistor</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L6"
                      >src/store.ts:6</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
        </div>
        <div class="col-4 col-menu menu-sticky-wrap menu-highlight">
          <nav class="tsd-navigation primary">
            <ul>
              <li class="">
                <a href="../modules.html">Modules</a>
              </li>
              <li class="current tsd-kind-module">
                <a href="store.html">store</a>
              </li>
            </ul>
          </nav>
          <nav class="tsd-navigation secondary menu-sticky">
            <ul class="before-current">
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#AppDispatch" class="tsd-kind-icon"
                  >App<wbr />Dispatch</a
                >
              </li>
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#RootState" class="tsd-kind-icon"
                  >Root<wbr />State</a
                >
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#default" class="tsd-kind-icon">default</a>
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#persistor" class="tsd-kind-icon"
                  >persistor</a
                >
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <footer class="with-border-bottom">
      <div class="container">
        <h2>Legend</h2>
        <div class="tsd-legend-group">
          <ul class="tsd-legend">
            <li class="tsd-kind-variable">
              <span class="tsd-kind-icon">Variable</span>
            </li>
            <li class="tsd-kind-function">
              <span class="tsd-kind-icon">Function</span>
            </li>
            <li class="tsd-kind-function tsd-has-type-parameter">
              <span class="tsd-kind-icon">Function with type parameter</span>
            </li>
            <li class="tsd-kind-type-alias">
              <span class="tsd-kind-icon">Type alias</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-enum">
              <span class="tsd-kind-icon">Enumeration</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-interface">
              <span class="tsd-kind-icon">Interface</span>
            </li>
          </ul>
        </div>
      </div>
    </footer>
    <div class="container tsd-generator">
      <p>
        Generated using
        <a href="https://typedoc.org/" target="_blank">TypeDoc</a>
      </p>
    </div>
    <div class="overlay"></div>
    <script src="../assets/js/main.js"></script>
  </body>
</html>
