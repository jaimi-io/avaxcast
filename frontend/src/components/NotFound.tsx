import { createStyles, makeStyles } from "@material-ui/core/styles";
import NothingHereLogo from "images/nothing_here.svg";
import NothingHereLogoBlack from "images/nothing_here_black.svg";
import SvgLogo from "./SvgLogo";

const useStyles = makeStyles(() =>
  createStyles({
    svg: {
      maxWidth: 700,
      maxHeight: 700,
      display: "flex",
      margin: "auto",
    },
  })
);

/**
 * Used for when data can be found i.e. 404 page
 * @returns The NotFound Component
 */
function NotFound(): JSX.Element {
  const classes = useStyles();
  return (
    <SvgLogo
      darkIcon={NothingHereLogo}
      lightIcon={NothingHereLogoBlack}
      className={classes.svg}
    />
  );
}

export default NotFound;
