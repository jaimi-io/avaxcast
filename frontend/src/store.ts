import { createStore } from "redux";
import { persistStore } from "redux-persist";
import rootReducer from "reducers/";

const store = createStore(rootReducer);
export const persistor = persistStore(store);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
