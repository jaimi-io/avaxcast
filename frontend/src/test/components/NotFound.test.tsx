import renderer from "react-test-renderer";
import NotFound from "components/NotFound";
import { Provider } from "react-redux";
import store from "store";
import { BrowserRouter as Router } from "react-router-dom";

test("NotFound snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <NotFound />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
