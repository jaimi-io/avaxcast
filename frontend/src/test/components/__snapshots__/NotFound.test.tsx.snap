// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshot 1`] = `
<img
  className="makeStyles-svg-1"
  src="nothing_here_black.svg"
/>
`;
